### DISCUSSION TITLE
Automazione - Farla partire una volta al giorno (e non di più)

### DISCUSSION THREAD

**Discussion by Borghese** [indomus/forum#2796] [GENERALE - Discussioni generiche]
Ciao ragazzi,
sono incastrato su un dubbio che non riesco a risolvere :
Ho creato un automazione che mi accende la pompa di calore per la produzione di acqua calda (comandata da uno shelly 1pm) se il fotovoltaico sta producendo più di quello che la casa sta consumando, con una soglia di immissione in rete di almeno 300w per 5 min consecutivi (questo lo monitoro con uno shelly em)
L'automazione di per se funziona, e ovviamente ho anche la gemella che mi spegne la pdc se il consumo della stessa è inferiore a 10w per almeno 1h (significa che ha terminato la produzione)

Il punto è che in questo modo, se all'interno della giornata succede che io sto immettendo almeno 300w in rete, l'automazione riparte ovviamente, facendomi riattaccare la pdc, che però ha l'accumulo di acqua già a temperatura, e quindi assorbe meno di 10w per 1h e re interviene l'automazione che la stacca

Per evitare questo stacca/attacca, io vorrei che l'automazione partisse solo una volta al giorno, ma non so come fare! 

Ho visto il comando pattern, ma non mi sembra si presti al caso mio, e anche la ricorrenza temporale non mi aiuta, perchè io non ho un orario fisso in cui farla partire, ma è legato alla produzione del fv e alle immissioni in rete.

Sapete come impostare una cosa del genere ? Un modo per dirgli... fai partire l'automazione, se OGGI non è già partita per esempio....

Questo è il codice dell'automazione attuale : 

```python
alias: Pdc Acs - On
description: ""
trigger:
  - type: power
    platform: device
    device_id: 00077fb1f27c7739eb14247f0467f452
    entity_id: 9c2aeebbfed21bb481fa77f26320e2ab
    domain: sensor
    below: -300
    for:
      hours: 0
      minutes: 5
      seconds: 0
condition:
  - condition: device
    type: is_off
    device_id: 066cdc707de38af3c465a98000fc96e7
    entity_id: e8384a953b2cc80912fda3b1e4348e47
    domain: switch
action:
  - type: turn_on
    device_id: 066cdc707de38af3c465a98000fc96e7
    entity_id: e8384a953b2cc80912fda3b1e4348e47
    domain: switch
  - service: notify.mobile_app_iphone_12_di_diego
    data:
      message: La produzione di acqua calda è iniziata.
  - service: notify.alexa_media
    data:
      message: La produzione di acqua calda è iniziata.
      target: media_player.in_tutta_la_casa
      data:
        type: announce
        method: spoken
mode: single
```

**Comment by Borghese** [comment]
Io sto continuando a provarci ma non trovo un modo per fare quanto descritto sopra....

**Reply by KC-inDomus** [reply]
non usare

```
  - type: power
    platform: device
    device_id: 00077fb1f27c7739eb14247f0467f452
    entity_id: 9c2aeebbfed21bb481fa77f26320e2ab
    domain: sensor
```

bensì platform: [numeric_state](https://indomus.it/formazione/esercizi-di-domotica-su-home-assistant-2-definire-diversi-trigger-per-innescare-automazioni/).

smettetela-di-farvi-le-automazioni-tramite-intefaccia-grafica

**Reply by Borghese** [reply]
Grazie @KC-inDomus ! Ma quindi è sufficiente usare platform : numeric_state per farla partire una volta al giorno ?!
